### PR TITLE
improve unifycr CLI launch tool

### PR DESCRIPTION
**Pull request by MichaelBrim**
### Description

Adds support for passing server CLI options, a method of
launch using a custom script, and full support for IBM
LSF/jsrun. Also implements placeholder functions for
terminate mode with other RMs. Note that the 
`--transfer-{in|out}` options have been renamed to 
`--stage-{in|out}`

This commit also removes stale files that are no longer
used with the recent changes in config settings management.

### How Has This Been Tested?

New jsrun start/stop functionality tested on Summitdev.

Will test on LLNL clusters with SLURM next, and will
create a separate PR for any needed fixes.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the UnifyCR code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted.

### CODE REVIEW

**Comment by adammoody** [comment]
Is this one ready to be merged?

My comment may not apply anymore, and if it does, we can come back to it in a later PR.

File: util/unifycr/src/unifycr-rm.c [hunk 1]
```diff
@@ -144,167 +144,554 @@ static int parse_hostfile(unifycr_resource_t *resource,
...
+    char *node;
+    char *last_node;
+    char *lsb_hosts;
+    char *pos;
+    char **nodes;
+    int mcpu = 0;
+
+    // LSB_HOSTS is space-separated list of host-slots with duplicates,
+    // and includes launch node as first entry
+    val = getenv("LSB_HOSTS");
```
[the remark this answers is not in the document]
Comment (MichaelBrim) [reply util/unifycr/src/unifycr-rm.c]: Yes, for larger jobs, LSB_HOSTS is not present in our environment. That's why there is a fallback to use LSB_MCPU_HOSTS. It's much more efficient to use the environment variable in memory than to open and parse the external file.

**Comment by MichaelBrim** [comment]
@adammoody This should be good to merge.

File: util/unifycr/src/unifycr-rm.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (adammoody) [reply util/unifycr/src/unifycr-rm.c]: Ok, perfect.  Sounds like IBM fixed it by dropping that variable for large jobs -- better than filling it with incomplete info.  I agree about using the envvar over reading a file.

**Pull request merged**